pyproject.toml

```diff
@@ -332,6 +332,17 @@ exclude = [
 
 [tool.ruff.lint.flake8-tidy-imports.banned-api]
 "urllib.request.urlopen".msg = "Use pandas.io.common.urlopen instead of urllib.request.urlopen"
+# numpy.random is banned but np.random is not. Is this intentional?
+# "numpy.random".msg = "Do not use numpy.random"
+"pytest.warns".msg = "Use tm.assert_produces_warning instead of pytest.warns"
+"pytest.xfail".msg = "Use pytest.mark.xfail instead of pytest.xfail"
+"conftest".msg = "No direct imports from conftest"
+"numpy.testing".msg = "Do not use numpy.testing"
+# "numpy.array_equal".msg = "Do not use numpy.array_equal" # Used in pandas/core
```

Comment:
Could we just ignore it's usage in pandas/core?

Reply:
AFAIK, We can only ignore `TID251` as a whole, we can't just ignore enforcing this rule for `numpy.array_equal` in `pandas/core`

Reply:
Gotcha. Can we remove some of the old checks that are covered by this section?

Reply:
it's done in the latest commit, you can take a look.